projects/project-2/readings/Project 2 - SPARQL library

```diff
@@ -0,0 +1,297 @@
+
+Title: Disjoint Subclass Conflict
+Constraint Description: Identifies classes that are marked as disjoint but share a common superclass.
+Severity: Error
+
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+SELECT ?class1 ?class2
+WHERE {
+  ?class1 owl:disjointWith ?class2 .
+  ?class1 rdfs:subClassOf ?commonSuperclass .
+  ?class2 rdfs:subClassOf ?commonSuperclass .
+}
+
+The goal of this query is to identify classes that are marked as disjoint but are simultaneously defined as subclasses of a common superclass.​ 
+This situation presents a logical inconsistency within the ontology, as disjoint classes should not share any common subclasses.
+
+5. Identifying Classes with Different Labeling Standards
+
+Title: Inconsistent Class Labeling
+Constraint Description: Detects classes with different labeling standards.
+Severity: Warning
+
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+SELECT ?class ?label1 ?label2
+WHERE {
+  ?class a owl:Class .
+  ?class rdfs:label ?label1 .
+  ?otherClass a owl:Class .
+  ?otherClass rdfs:label ?label2 .
+  FILTER(?class != ?otherClass) 
+  FILTER(?label1 != ?label2)
+}
+
+The goal of this query is to find classes within an ontology that are associated with different labeling standards.​ 
+This can be important for identifying inconsistencies in how classes are labeled or described, which can help improve the quality and usability of the ontology.
+
+6. Identifying Potentially Overspecified Class Definitions
+
+Title: Overspecified Class Detection
+Constraint Description: Identifies classes with potentially redundant or overlapping property definitions.
+Severity: Warning
+
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+SELECT ?class ?property1 ?property2 ?definition1 ?definition2
+WHERE {
+  ?class a owl:Class .
+  ?class ?property1 ?definition1 .
+  ?class ?property2 ?definition2 .
+  FILTER(?property1 != ?property2) 
+}
+
+The aim of this query is to identify classes within an ontology that may have overspecified definitions.​ 
+This occurs when a class is defined with multiple properties that provide overlapping or redundant constraints, potentially leading to confusion or misinterpretation of the class's intent.
+
+7. Detecting Equivalent Properties with Different Ranges
+
+Title: Inconsistent Equivalent Property Ranges
+Constraint Description: Identifies equivalent properties with different range specifications.
+Severity: Error
+
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+SELECT ?property1 ?property2 ?range1 ?range2
+WHERE {
+  ?property1 a owl:ObjectProperty .
+  ?property2 a owl:ObjectProperty .
+  ?property1 owl:equivalentProperty ?property2 .
+  ?property1 rdfs:range ?range1 .
+  ?property2 rdfs:range ?range2 .
+  FILTER(?range1 != ?range2)
+}
+
```

Comment:
It seems like the query only considers properties that are explicitly declared as equivalent using owl:equivalentProperty. It will not catch cases where properties might be inferred as equivalent through other means (e.g., transitivity of equivalence) unless reasoning is enabled and the RDF store materializes those inferences. But I don't think that's a big problem.